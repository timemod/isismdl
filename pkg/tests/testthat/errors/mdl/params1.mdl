param c1 1;
param c2 1 2 3;

function f1(x) = sum(j = -2, -1 : x(j));
function f2(x) = sum(j = 1, 2 : x(j));
function f3(x) = sum(j = 1, 2 : x(j-2));
function f4(x) = x(-3);
function f5(x) = f1(x);
function f6(x, p) = p * f1(x(3));

ident eq1   x1 = c2(-2);
ident eq2   x2 = c2(-4);
ident eq3   x3 = c2(2);
ident eq4   x4 = c1(-1);
ident eq5   x5 = f1(c1);
ident eq6   x6 = f1(c2);
ident eq7   x7 = f2(c2);
ident eq8   x8 = f2(c2(-3));
ident eq9   x9 = f3(c2);
ident eq10 x10 = f4(c1);
ident eq11 x11 = f4(c2);
ident eq12 x12 = f5(c2(-1));
ident eq13 x13 = f6(c1, 1) + c1(-2);
ident eq14 x14 = f6(c2, 2);
ident eq15 x15 = f6(c2(-3), 3);
ident eq16 x16 = sum(j = -2, 0 : c1(j));
ident eq17 x17 = sum(j = -2, 0 : c2(j));
