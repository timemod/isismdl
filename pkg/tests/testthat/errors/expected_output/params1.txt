Invalid index (-4) on param c2
The index should obey -2 <= index <= 0
Line: 12 of file mdl/params1.mdl
ident eq2   x2 = c2(-4);
                      ^

Invalid index (2) on param c2
The index should obey -2 <= index <= 0
Line: 13 of file mdl/params1.mdl
ident eq3   x3 = c2(2);
                     ^

Invalid index (-1) on param c1
The index should obey 0 <= index <= 0
Line: 14 of file mdl/params1.mdl
ident eq4   x4 = c1(-1);
                      ^

Invalid index range (-2:-1) on param c1 due to function call 
The index should obey 0 <= index <= 0
Line: 15 of file mdl/params1.mdl
ident eq5   x5 = f1(c1);
                      ^

Invalid index range (1:2) on param c2 due to function call 
The index should obey -2 <= index <= 0
Line: 17 of file mdl/params1.mdl
ident eq7   x7 = f2(c2);
                      ^

Invalid index (-3) on param c1 due to function call 
The index should obey 0 <= index <= 0
Line: 20 of file mdl/params1.mdl
ident eq10 x10 = f4(c1);
                      ^

Invalid index (-3) on param c2 due to function call 
The index should obey -2 <= index <= 0
Line: 21 of file mdl/params1.mdl
ident eq11 x11 = f4(c2);
                      ^

Invalid index range (-3:-2) on param c2 due to function call 
The index should obey -2 <= index <= 0
Line: 22 of file mdl/params1.mdl
ident eq12 x12 = f5(c2(-1));
                          ^

Invalid index range (1:2) on param c1 due to function call 
The index should obey 0 <= index <= 0
Line: 23 of file mdl/params1.mdl
ident eq13 x13 = f6(c1, 1) + c1(-2);
                         ^

Invalid index (-2) on param c1
The index should obey 0 <= index <= 0
Line: 23 of file mdl/params1.mdl
ident eq13 x13 = f6(c1, 1) + c1(-2);
                                  ^

Invalid index range (1:2) on param c2 due to function call 
The index should obey -2 <= index <= 0
Line: 24 of file mdl/params1.mdl
ident eq14 x14 = f6(c2, 2);
                         ^

Invalid index range (-2:0) on param c1
The index should obey 0 <= index <= 0
Line: 26 of file mdl/params1.mdl
ident eq16 x16 = sum(j = -2, 0 : c1(j));
                                     ^

